Build failure: jamesdsp

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *jamesdsp*

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: qmakePrePhase
@nix { "action": "setPhase", "phase": "qmakePrePhase" }
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/7z6vyin837llzfswjg877av13fg7zgrz-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
QMAKEPATH=/nix/store/wbnjx3rfynzl533732sjwj5cs1ix65vg-qtbase-6.8.1-dev:/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1:/nix/store/2by3b885r305iwrpyy7gzskam6ks8li3-qtsvg-6.8.1-dev:/nix/store/02cpzfrnn16sm8q1xm3cg34scsrb2rfz-qtwayland-6.8.1-dev:/nix/store/wr239a7nr2zcnm5w6kb4gzdk15x548qj-qtdeclarative-6.8.1-dev:/nix/store/8vl6yaccvg97lnhzg6dg6jnjb0yx0da6-qtlanguageserver-6.8.1-dev:/nix/store/93bhmdrbmn6dha8dqrljyh4ag7h2d3kr-qtshadertools-6.8.1-dev:/nix/store/kfacgz2mdq6dfzpyiih9qq9x8n6cmail-qtdeclarative-6.8.1
qmake PREFIX=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0 NIX_OUTPUT_OUT=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0 NIX_OUTPUT_DEV=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0 NIX_OUTPUT_BIN=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0 NIX_OUTPUT_QML=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0/lib/qt-6/qml NIX_OUTPUT_PLUGIN=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0/lib/qt-6/plugins
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
Info: creating stash file /build/source/.qmake.stash
qmake: enabled parallel building
qmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16 SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
cd libjamesdsp/ && ( test -e Makefile || /nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/bin/qmake -o Makefile /build/source/libjamesdsp/libjamesdsp.pro PREFIX=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0 NIX_OUTPUT_OUT=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0 NIX_OUTPUT_DEV=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0 NIX_OUTPUT_BIN=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0 NIX_OUTPUT_QML=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0/lib/qt-6/qml NIX_OUTPUT_PLUGIN=/nix/store/y2vaqc1in3yrc7xwzx01sbb75nw4davf-jamesdsp-2.7.0/lib/qt-6/plugins ) && make -f Makefile 
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
make[1]: Entering directory '/build/source/libjamesdsp'
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o arbEqConv.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/arbEqConv.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o bs2b.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/bs2b.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o convolver1D.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/convolver1D.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o crossfeed.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/crossfeed.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o dbb.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/dbb.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o dynamic.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/dynamic.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o cpthread.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/cpthread.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o fft.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/fft.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o nseel-compiler.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o nseel-ram.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-ram.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o FFTConvolver.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/numericSys/FFTConvolver.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o cos_fib_paraunitary.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/numericSys/FilterDesign/cos_fib_paraunitary.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o eqnerror.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/numericSys/FilterDesign/eqnerror.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o firls.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/numericSys/FilterDesign/firls.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o generalFdesign.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/numericSys/FilterDesign/generalFdesign.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o polyphaseASRC.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/numericSys/FilterDesign/polyphaseASRC.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o polyphaseFilterbank.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/numericSys/FilterDesign/polyphaseFilterbank.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o MersenneTwister.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/numericSys/MersenneTwister.c
gcc -c -pipe -std=gnu11 -O2 -O2 -fPIC -w -D_REENTRANT -DLIBJAMESDSP_PLUGIN -DCUSTOM_CMD -Dprintf=redirected_printf -DANDROID_LOG_INFO=0 -DDEBUG -DQT_NO_DEBUG -DQT_CORE_LIB -I. -Isubtree/Main/libjamesdsp/jni/jamesdsp/jdsp -Isubtree/Main/libjamesdsp/jni/jamesdsp -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/include/QtCore -I. -I/nix/store/dh5kclrfrvk7aknvk22n2bnz4x8kj1q6-qtbase-6.8.1/mkspecs/linux-g++ -o inv.o subtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/numericSys/SolveLinearSystem/inv.c
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/crossfeed.c:[m[K In function '[01m[KCrossfeedEnable[m[K':
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/crossfeed.c:65:45:[m[K [01;31m[Kerror: [m[Kassignment to '[01m[Kvoid (*)(struct dspsys *, size_t)[m[K' {aka '[01m[Kvoid (*)(struct dspsys *, long unsigned int)[m[K'} from incompatible pointer type '[01m[Kvoid (*)(JamesDSPLib *, size_t)[m[K' {aka '[01m[Kvoid (*)(struct dspsys *, long unsigned int)[m[K'} [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
   65 |                         jdsp->advXF.process [01;31m[K=[m[K CrossfeedProcessFFTConvolver2x4x2;
      |                                             [01;31m[K^[m[K
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/crossfeed.c:73:45:[m[K [01;31m[Kerror: [m[Kassignment to '[01m[Kvoid (*)(struct dspsys *, size_t)[m[K' {aka '[01m[Kvoid (*)(struct dspsys *, long unsigned int)[m[K'} from incompatible pointer type '[01m[Kvoid (*)(JamesDSPLib *, size_t)[m[K' {aka '[01m[Kvoid (*)(struct dspsys *, long unsigned int)[m[K'} [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
   73 |                         jdsp->advXF.process [01;31m[K=[m[K CrossfeedProcessTwoStageFFTConvolver2x4x2;
      |                                             [01;31m[K^[m[K
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/crossfeed.c:[m[K In function '[01m[KCrossfeedProcess[m[K':
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/crossfeed.c:124:37:[m[K [01;31m[Kerror: [m[Kpassing argument 1 of '[01m[Kjdsp->advXF.process[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  124 |                 jdsp->advXF.process([01;31m[Kjdsp[m[K, (unsigned int)n);
      |                                     [01;31m[K^~~~[m[K
      |                                     [01;31m[K|[m[K
      |                                     [01;31m[KJamesDSPLib * {aka struct dspsys *}[m[K
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/crossfeed.c:124:37:[m[K [01;36m[Knote: [m[Kexpected '[01m[Kstruct dspsys *[m[K' but argument is of type '[01m[KJamesDSPLib *[m[K' {aka '[01m[Kstruct dspsys *[m[K'}
make[1]: *** [Makefile:1047: crossfeed.o] Error 1
make[1]: *** Waiting for unfinished jobs....
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:[m[K In function '[01m[Kkth_smallest[m[K':
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:1649:18:[m[K [01;31m[Kerror: [m[Ktype defaults to '[01m[Kint[m[K' in declaration of '[01m[Ki[m[K' [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int]8;;[m[K]
 1649 |         register [01;31m[Ki[m[K, j, l, m;
      |                  [01;31m[K^[m[K
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:1649:21:[m[K [01;31m[Kerror: [m[Ktype defaults to '[01m[Kint[m[K' in declaration of '[01m[Kj[m[K' [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int]8;;[m[K]
 1649 |         register i, [01;31m[Kj[m[K, l, m;
      |                     [01;31m[K^[m[K
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:1649:24:[m[K [01;31m[Kerror: [m[Ktype defaults to '[01m[Kint[m[K' in declaration of '[01m[Kl[m[K' [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int]8;;[m[K]
 1649 |         register i, j, [01;31m[Kl[m[K, m;
      |                        [01;31m[K^[m[K
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:1649:27:[m[K [01;31m[Kerror: [m[Ktype defaults to '[01m[Kint[m[K' in declaration of '[01m[Km[m[K' [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int]8;;[m[K]
 1649 |         register i, j, l, [01;31m[Km[m[K;
      |                           [01;31m[K^[m[K
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:[m[K In function '[01m[K_eel_flacDecodeFile[m[K':
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:3859:92:[m[K [01;31m[Kerror: [m[Kpassing argument 4 of '[01m[Kdrflac_open_file_and_read_pcm_frames_f32[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 3859 |         float *signal = drflac_open_file_and_read_pcm_frames_f32(filename, &channels, &fs, [01;31m[K&frameCount[m[K, 0);
      |                                                                                            [01;31m[K^~~~~~~~~~~[m[K
      |                                                                                            [01;31m[K|[m[K
      |                                                                                            [01;31m[Kuint64_t * {aka long unsigned int *}[m[K
In file included from [01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:3849[m[K:
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/dr_flac.h:11658:147:[m[K [01;36m[Knote: [m[Kexpected '[01m[Kdrflac_uint64 *[m[K' {aka '[01m[Klong long unsigned int *[m[K'} but argument is of type '[01m[Kuint64_t *[m[K' {aka '[01m[Klong unsigned int *[m[K'}
11658 | DRFLAC_API float* drflac_open_file_and_read_pcm_frames_f32(const char* filename, unsigned int* channels, unsigned int* sampleRate, [01;36m[Kdrflac_uint64* totalPCMFrameCount[m[K, const drflac_allocation_callbacks* pAllocationCallbacks)
      |                                                                                                                                    [01;36m[K~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~[m[K
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:[m[K In function '[01m[K_eel_flacDecodeMemory[m[K':
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/nseel-compiler.c:3921:107:[m[K [01;31m[Kerror: [m[Kpassing argument 5 of '[01m[Kdrflac_open_memory_and_read_pcm_frames_f32[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 3921 |         float *signal = drflac_open_memory_and_read_pcm_frames_f32(memoryBlk, actualSize, &channels, &fs, [01;31m[K&frameCount[m[K, 0);
      |                                                                                                           [01;31m[K^~~~~~~~~~~[m[K
      |                                                                                                           [01;31m[K|[m[K
      |                                                                                                           [01;31m[Kuint64_t * {aka long unsigned int *}[m[K
[01m[Ksubtree/Main/libjamesdsp/jni/jamesdsp/jdsp/Effects/eel2/dr_flac.h:11725:162:[m[K [01;36m[Knote: [m[Kexpected '[01m[Kdrflac_uint64 *[m[K' {aka '[01m[Klong long unsigned int *[m[K'} but argument is of type '[01m[Kuint64_t *[m[K' {aka '[01m[Klong unsigned int *[m[K'}
11725 | DRFLAC_API float* drflac_open_memory_and_read_pcm_frames_f32(const void* data, size_t dataSize, unsigned int* channels, unsigned int* sampleRate, [01;36m[Kdrflac_uint64* totalPCMFrameCount[m[K, const drflac_allocation_callbacks* pAllocationCallbacks)
      |                                                                                                                                                   [01;36m[K~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~[m[K
make[1]: *** [Makefile:1119: nseel-compiler.o] Error 1
make[1]: Leaving directory '/build/source/libjamesdsp'
make: *** [Makefile:48: sub-libjamesdsp-make_first] Error 2

```

</details>

## Additional context

This only started to manifest from the latest revision of `nixpkgs-unstable`. The latest version number of the package on said channel is `2.7.0`.

## Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.5, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`
```

## Notify maintainers

@pasqui23 @wineee 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/audio/jamesdsp/default.nix
+++ b/pkgs/applications/audio/jamesdsp/default.nix
@@ -66,6 +66,13 @@ stdenv.mkDerivation (finalAttrs: {
 
   qmakeFlags = lib.optionals usePulseaudio [ "CONFIG+=USE_PULSEAUDIO" ];
 
+  # https://github.com/Audio4Linux/JDSP4Linux/issues/228
+  env.NIX_CFLAGS_COMPILE = toString [
+    "-Wno-error=incompatible-pointer-types"
+    "-Wno-error=implicit-int"
+    "-Wno-error=implicit-function-declaration"
+  ];
+
   desktopItems = [
     (makeDesktopItem {
       name = "jamesdsp";

